### PR TITLE
Only import types where possible, and add consistent-type-imports rule

### PR DESCRIPTION
**Pull request by birkskyum**
If only the type is used, there's no need to import the value of the classes etc.

This adds an eslint rule for it.

## Launch Checklist

 - [ ] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [ ] Link to related issues.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: src/data/bucket/fill_extrusion_bucket.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import {FillExtrusionLayoutArray, PosArray} from '../array_types.g';
 
 import {members as layoutAttributes, centroidAttributes} from './fill_extrusion_attributes';
-import {Segment, SegmentVector} from '../segment';
+import type {Segment} from '../segment';
```
Comment (HarelM) [inline src/data/bucket/fill_extrusion_bucket.ts:4]: Is there a way to configure this rule to make a single import line for these kind of situations?

File: eslint.config.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ export default [
         rules: {
             'no-dupe-class-members': 'off',
             '@typescript-eslint/no-dupe-class-members': ['error'],
-
+            '@typescript-eslint/consistent-type-imports': ['error'],
```
Comment (HarelM) [inline eslint.config.js:46]: Is it possible to add a rule to sort imports based on "from" alphabetically?
It works remove the bed to think where to add the import at I think.

Comment (birkskyum) [reply eslint.config.js]: It's possible to configure to prefer inline, which I've done ow

File: src/data/bucket/fill_extrusion_bucket.ts (hunk 1 above)
Comment (birkskyum) [reply src/data/bucket/fill_extrusion_bucket.ts]: It's possible to configure to prefer inline, which I've done now,

**Review by HarelM — APPROVED** [review]
Assuming this was done automatically... So approved.

File: eslint.config.js (hunk 2 above)
Comment (HarelM) [reply eslint.config.js]: It won't, but lint will just make sure the imports are ordered correctly, so there's no need to ever touch that part of the file.

Comment (birkskyum) [reply eslint.config.js]: just realized sort-imports are sorting on the values of the vars imported, no the path they're imported from

**Pull request merged**